util/fibers/detail/fiber_interface.cc

```diff
@@ -267,24 +266,13 @@ ctx::fiber_context FiberInterface::SwitchTo() {
   });
 }
 
-void FiberInterface::PrintStackTrace() {
-  std::string_view state = "sleeping";
-  if (list_hook.is_linked()) {
-    state = "ready";
-  } else if (FiberActive() == this) {
-    state = "active";
-  }
-
-  auto print = [this, state]() {
-    LOG(INFO) << "------------ Fiber " << this->name_ << " (" << state << ") ------------\n" << GetStacktrace();
-  };
-
+void FiberInterface::ExecuteOnFiberStack(PrintFn fn) {
   if (FiberActive() == this) {
-    return print();
+    return fn(this);
   }
 
-  entry_ = std::move(entry_).resume_with([print = std::move(print)](ctx::fiber_context&& c) {
-    print();
+  entry_ = std::move(entry_).resume_with([fn = std::move(fn), this](ctx::fiber_context&& c) {
```

Comment:
Is this comprehensible?

```suggestion
  // We're in a random fiber but we want to execute in the context of `this`. We call
  // `this->entry_.resume_with(L)`. Calling this method suspends the current fiber and executes `L`
  // on top of the stack of `this->entry_`.
  // It is similar to running an interrupt handler in the same stack as a suspended program.
  // Inside `L`, we execute `fn` and then resume the original fiber that was suspended which is
  // passed to us as the argument. When the other fiber will be resumed normally, it will execute
  // the final `return` statement and go back to its original suspension site.
  entry_ = std::move(entry_).resume_with([fn = std::move(fn), this](ctx::fiber_context&& c) {
```

Reply:
site -> state.